hedera-node/hedera-app/src/main/java/com/hedera/node/app/grpc/impl/QueryMethod.java

```diff
@@ -51,6 +51,11 @@ public final class QueryMethod extends MethodBase {
     /** A metric for the calls per second that queries were answered */
     private final SpeedometerMetric queriesAnsweredSpeedometer;
 
+    /**
+     * Indicates if the QueryWorkflow should charge for handling queries.
+     */
+    private final boolean shouldCharge;
```

Comment:
Is there a reason to make this a per-`QueryMethod` field instead of using an all-or-nothing config inside `QueryWorkflow`?

(I.e., would it make sense to charge for some methods and not others?)

Reply:
> implementations with @UserQueries and @OperatorQueries qualifiers instead of a @Singleton...might be cleaner, not sure.

Agreed